app/controllers/passengers_controller.rb

```diff
@@ -0,0 +1,66 @@
+class PassengersController < ApplicationController
+  def index
+    if params[:search]
+      @passengers = Passenger.search(params[:search]).paginate(page: params[:page], per_page: 10)
+    else
+      @passengers = Passenger.all.paginate(page: params[:page], per_page: 10)
+    end
+  end
+
+  def show
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+    end
+  end
+
+  def new
+    @passenger = Passenger.new
+  end
+
+  def create
+    @passenger = Passenger.new(passenger_params)
+
+    if @passenger.save
+      redirect_to passenger_path(@passenger)
+    else
+      render :new, status: :bad_request
```

Comment:
Wonderful controller code! I love rendering a view and sending a status with it! That's the thoroughness and detail I like!